lib/posix/signal.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "signal.h"
+
+#include <errno.h>
+#include <string.h>
+
+#include <zephyr/posix/signal.h>
+
+#define SIGNO_WORD_IDX(_signo) ((signo - 1) / _NSIG_BPW)
+#define SIGNO_WORD_BIT(_signo) ((signo - 1) % _NSIG_BPW)
```

Comment:
```suggestion
#define SIGNO_WORD_IDX(_signo) (signo / BITS_PER_LONG)
#define SIGNO_WORD_BIT(_signo) (signo & BIT_MASK(BITS_PER_LONG))
```

Reply:
I think the latter `&` might be wrong - it should be something like `BIT_MASK(LOG2(BITS_PER_LONG))`